src/Rings/mpoly-affine-algebras.jl

```diff
@@ -153,9 +153,6 @@ function monomial_basis(A::MPolyQuoRing)
   is_finite_dimensional_vector_space(A) || throw(InfiniteDimensionError())
   I = A.I
   G = standard_basis(I)
-  if dim(I) == -inf # I is the whole ring
```

Comment:
@joschmitt you introduced this check in PR #3497 but the comment here doesn't seem to match the code, and in any case I was not able to trigger this check: if `I` is the zero ideal then `A` is infinite dimensional and we error out a few lines before. Well, unless the underlying polynomial ring is trivial (has zero generators), but in that case many other things break down (such as calling `is_trivial(A)` to check for this situations...)

Reply:
If `dim(I) == -inf`, then the quotient is the zero ring. The comment says `I` is the whole ring, not `A`. :)
You can trigger it with
```
julia> R, x = QQ["x1", "x2"]
(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x1, x2])

julia> A, _ = quo(R, ideal(R(1)))
(Quotient of multivariate polynomial ring by ideal (1), Map: R -> A)

julia> monomial_basis(A)
QQMPolyRingElem[]
```

Reply:
Thank you @joschmitt I had a "brain fart", somehow my brain turned "zero polynomials have degree -inf" into "zero ideals have dim -inf", which of course is utter nonsense. I'll revise this PR.